h/api/views.py

```diff
@@ -201,20 +202,37 @@ def read_jsonld(annotation, request):
     return presenter.asdict()
 
 
-@api_config(route_name='api.annotation', request_method='PUT', permission='update')
+@api_config(route_name='api.annotation',
+            request_method='PUT',
+            permission='update')
 def update(annotation, request):
     """Update the specified annotation with data from the PUT payload."""
-    schema = schemas.LegacyUpdateAnnotationSchema(request,
-                                                  annotation=annotation)
-    appstruct = schema.validate(_json_payload(request))
-    annotation = storage.legacy_update_annotation(request,
-                                                  annotation.id,
-                                                  appstruct)
+    json_payload = _json_payload(request)
 
-    _publish_annotation_event(request, annotation, 'update')
+    legacy_annotation = elastic.Annotation.fetch(annotation.id)
```

Comment:
Ok, this function was a mess, I've refactored it in a way that should make the code a lot clearer without changing the behaviour. Also stopped it from unnecessarily fetching the legacy annotation when postgres if off (in that case the legacy annotation is the one already passed to the view as argument). And called fetch_annotation() instead of models.elastic when it does need to fetch the legacy annotation, as you suggested.